api/src/auth_routes.ts

```diff
@@ -395,12 +393,10 @@ export function add_auth_routes(app: Router, handlers: string[]) {
       `/auth-return/${handler}/`,
       (req: any, res: any, next: any) => {
         const redirect = req.session['redirect'] || '/send-token';
-        passport.authenticate(handler + '-validate', {
-          successRedirect: redirect,
-          failureRedirect: '/auth',
-          failureFlash: true,
-          successFlash: 'Welcome!',
-        })(req, res, next);
+        passport.authenticate(
+          handler + '-validate',
+          authenticate_return(req, res, next, redirect)
```

Comment:
I note that this authenticate_Return function does not distinguish between successful and unsuccessful redirects where as the previous version does. I also note the error messages in the authenticate_return function don't make sense in the context of a social sign in e.g. "invalid email or password" - this could be confusing to the user.